vendor/knative.dev/eventing/pkg/reconciler/source/crd/controller.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/eventing/pkg/apis/sources"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	pkgreconciler "knative.dev/pkg/reconciler"
+
+	crdinfomer "knative.dev/pkg/client/injection/apiextensions/informers/apiextensions/v1/customresourcedefinition"
+	crdreconciler "knative.dev/pkg/client/injection/apiextensions/reconciler/apiextensions/v1/customresourcedefinition"
+)
+
+const (
+	// ReconcilerName is the name of the reconciler.
+	ReconcilerName = "SourceCRDs"
+)
+
+// NewController creates a Reconciler for Sources' CRDs and returns the result of NewImpl.
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+	crdInformer := crdinfomer.Get(ctx)
+
+	r := &Reconciler{
+		ogctx:       ctx,
+		ogcmw:       cmw,
+		controllers: make(map[schema.GroupVersionResource]runningController),
+	}
+	filterFunc := pkgreconciler.LabelFilterFunc(sources.SourceDuckLabelKey, sources.SourceDuckLabelValue, false)
```

Comment:
Looking at this filter, it should work by changing the label key on the KafkaSource CRD.

https://github.com/knative-sandbox/eventing-kafka/blob/7a13f9656bd7aef3fe07ccfbf58a7fb509726b05/config/source/common/resources/source/kafkasource.yaml#L20

that should be `sources.knative.dev/source: "true"`

Reply:
Label key: https://github.com/knative-sandbox/eventing-kafka/blob/2b185a9b59347cc4a16471ed0fb2ba3bdf6e1672/vendor/knative.dev/eventing/pkg/apis/sources/register.go#L30

Reply:
look here:
https://github.com/knative/eventing/blob/main/config/core/resources/pingsource.yaml#L20

they all use `eventing.knative.dev/source` and the spec just mentions the `duck`:
https://github.com/knative/eventing/blob/main/docs/spec/sources.md#source-crds

Reply:
so even I change that... I do not see any event type.... 

broken API ... :see_no_evil:

Reply:
this is messy. The aggregates ClusterRole expects `duck.knative.dev/source: "true"`

Reply:
we have `duck.knative.dev/source: "true"` already - today:
https://github.com/knative-sandbox/eventing-kafka/blob/main/config/source/common/resources/source/kafkasource.yaml#L21